### PR TITLE
tchau-exercises

### PR DESCRIPTION
**Pull request by tranconghau**
This pull-request is just only tracking the commit easier.
Please do not merge it into the master branch. Thanks. :dancer:

### CODE REVIEW

File: tchau/array/array_ex1.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+int findMissedNumberInIntegerArray(int arr[], int min, int max)
+{
+	int sum = 0;
+	int tmpSum = 0;
+	for(int i=min; i<=max; i++)
+	{
+		tmpSum+=i;
+		sum+=arr[i-1];
+#if DEBUG
```
Comment (maixuanhan) [inline tchau/array/array_ex1.cpp]: Avoid this kind of check in simple programs, functions. A macro function maybe used for printing such debug information.

File: tchau/array/array_ex1.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		cout<<"We only accept the number from "<<MIN<<" to "<<MAX<<endl;
+		return -1;
+	}
+
+	memset(arr, 0, MAX);
+	for (int i = MIN; i <= MAX; ++i)
+	{
+		if (i != num)
+		{
+			arr[i-1] = i;
```
Comment (maixuanhan) [inline tchau/array/array_ex1.cpp]: If `MIN` value is changed (such as 0), your code will be broken.

File: tchau/array/array_ex1.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#if DEBUG
+			cout<<"Added a["<<i-1<<"] = "<<arr[i-1]<<endl;
+#endif
+		}
+	}
+
+	ret = findMissedNumberInIntegerArray(arr, MIN, MAX);
+	cout<< "Your array has missed: "<< ret<< endl;
+	return 0;
+}
```
Comment (maixuanhan) [inline tchau/array/array_ex1.cpp]: For good practice, have new line before EOF :)

File: tchau/array/array_ex1.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	cout<<"Sum: "<<sum<<endl;
+	cout<<"tmpSum: "<<tmpSum<<endl;
+#endif
+
+	return tmpSum - sum;
+}
+
+
+
+int main(int argc, char const *argv[])
```
Comment (maixuanhan) [inline tchau/array/array_ex1.cpp:58]: simple testcases (hard coded or generated) without user interaction is preferred for this kind of UT test

**Review by maixuanhan — CHANGES_REQUESTED** [review]

**Comment by maixuanhan** [comment]
> This pull-request is just only tracking the commit easier.
> Please do not merge it into the master branch. Thanks.

It will be merged eventually.

File: tchau/array/array_ex1.cpp (hunk 3 above)
Comment (tranconghau) [reply tchau/array/array_ex1.cpp]: Will be fixed in the next time! :dancer:

File: tchau/array/array_ex1.cpp (hunk 2 above)
Comment (tranconghau) [reply tchau/array/array_ex1.cpp]: The program has been changed to follow the requirement. It also fixes this issue :)

**Review by maixuanhan — APPROVED** [review]

**Pull request merged**